kuksa_databroker/databroker/src/glob.rs

```diff
@@ -51,3 +52,113 @@ pub fn to_regex(glob: &str) -> Result<Regex, Error> {
     let re = to_regex_string(glob);
     Regex::new(&re).map_err(|_err| Error::RegexError)
 }
+
+fn has_asterisk_in_middle(input: &str) -> bool {
+    let input = input.trim(); // Remove leading and trailing whitespace
+    if input.starts_with('*') || input.ends_with('*') {
+        return false;
+    }
+
+    let asterisk_position = input.find('*');
+    if let Some(pos) = asterisk_position {
+        if input[pos - 1..].contains(".*.") {
+            return true;
+        }
+    }
+
+    false
+}
+
+pub fn to_regex_string_new(glob: &str) -> String {
+    let mut input = String::from(glob);
+
+    //check if asterisk in middle
+    if has_asterisk_in_middle(glob) {
+        input = glob.replace(".*.", ".");
+    }
+    // Construct regular expression
+
+    // Make sure we match the whole thing
+    // Start from the beginning
+    let mut re = String::from("^");
+
+    if input.ends_with(".*") {
+        re.push_str(input.replace(".*", ".[^.]+").as_str());
+    } else if input.ends_with(".**") {
```

Comment:
If it ends with `.**` shouldn't other replacements be done anyway?

Reply:
Same here, following the use case 'Vehicle.Cabin.Sunroof.\*\*', must be replaced with '.\*', so it returns any possible child under it.
The reason for these two if else statements is that the split and join per '.' algorithm implemented below doesn't cover the use cases than end with '.\*' or '.\*\*' adding extra '\\'